control-plane/cni/main.go

```diff
@@ -220,15 +230,21 @@ func (c *Command) cmdAdd(args *skel.CmdArgs) error {
 		return fmt.Errorf("could not apply iptables setup: %v", err)
 	}
 
-	// We do not throw an error here because kubernetes will often throw a benign error where the pod has been
-	// updated in between the get and update of the annotation. Eventually kubernetes will update the annotation
-	ok = c.updateTransparentProxyStatusAnnotation(podName, podNamespace, complete)
-	if !ok {
-		logger.Info("unable to update %s pod annotation to complete", keyTransparentProxyStatus)
+	if cniArgsIPTablesCfg == "" {
```

Comment:
Maybe somewhere near the top we can add a comment that if `cniArgsIPTablesCfg` is empty then we are on K8s and otherwise we are on Nomad

Reply:
Good call, will do.